tests/unit/modules/tls_test.py

```diff
@@ -0,0 +1,623 @@
+                             _TLS_TEST_DATA['ca_cert'])
+
+    @patch('os.path.exists', MagicMock(return_value=True))
+    @patch('salt.modules.tls.maybe_fix_ssl_version',
+           MagicMock(return_value=True))
+    def test_get_ca(self):
+        '''
+        Test get_ca
+        '''
+        ca_path = '/tmp/test_tls'
+        ca_name = 'test_ca'
+        certp = '{0}/{1}/{2}_ca_cert.crt'.format(
+            ca_path,
+            ca_name,
+            ca_name)
+        mock = MagicMock(return_value=ca_path)
+        with patch.dict(tls.__salt__, {'config.option': mock}):
+            self.assertEqual(tls.get_ca(ca_name), certp)
+
+    @patch('os.path.exists', MagicMock(return_value=True))
+    @patch('salt.modules.tls.maybe_fix_ssl_version',
+           MagicMock(return_value=True))
+    @patch('salt.utils.fopen', mock_open(read_data=_TLS_TEST_DATA['ca_cert']))
+    def test_cert_info(self):
+        '''
+        Test cert info
+        '''
+        ca_path = '/tmp/test_tls'
+        ca_name = 'test_ca'
+        certp = '{0}/{1}/{2}_ca_cert.crt'.format(
+            ca_path,
+            ca_name,
+            ca_name)
+        ret = {
+            'not_after': 1462379961.0,
+            'signature_algorithm': 'sha256WithRSAEncryption',
+            'extensions': None,
+            'fingerprint': ('96:72:B3:0A:1D:34:37:05:75:57:44:7E:08:81:A7:09:'
+                            '0C:E1:8F:5F:4D:0C:49:CE:5B:D2:6B:45:D3:4D:FF:31'),
+            'serial_number': 284092004844685647925744086791559203700L,
+            'subject': {
+                'C': 'US',
+                'CN': 'localhost',
+                'L': 'Salt Lake City',
+                'O': 'SaltStack',
+                'ST': 'Utah',
+                'emailAddress':
+                'xyz@pdq.net'},
+            'not_before': 1430843961.0,
+            'issuer': {
+                'C': 'US',
+                'CN': 'localhost',
+                'L': 'Salt Lake City',
+                'O': 'SaltStack',
+                'ST': 'Utah',
+                'emailAddress': 'xyz@pdq.net'}
+        }
+
+        def ignore_extensions(data):
+            '''
+            Ignore extensions pending a resolution of issue 24338
+            '''
+            if 'extensions' in data.keys():
+                data['extensions'] = None
+            return data
+
+        # older pyopenssl versions don't have extensions or
+        # signature_algorithms
+        def remove_not_in_result(source, reference):
+            if 'signature_algorithm' not in reference:
+                del source['signature_algorithm']
+            if 'extensions' not in reference:
+                del source['extensions']
+        result = ignore_extensions(tls.cert_info(certp))
+        remove_not_in_result(ret, result)
+        self.assertEqual(result, ret)
+
+    @patch('salt.modules.tls.maybe_fix_ssl_version',
+           MagicMock(return_value=True))
+    def test_create_ca(self):
```

Comment:
No, `destructiveTest` means that the test will touch the system outside of the test runner and could potentially cause harm to the system even though the test makes a best effort to clean up after itself.  `destructiveTest`s are protected in the test runner and can only be run when explicitly enabled, `python tests/runtests.py -u --run-destructuve`, for example.